lib/pbench/test/functional/server/conftest.py

```diff
@@ -24,33 +31,35 @@ def server_client():
     return client
 
 
-@pytest.fixture(scope="module")
-def register_test_user(server_client: PbenchServerClient):
-    """Create a test user for functional tests."""
+@pytest.fixture(scope="session")
+def oidc_admin(server_client: PbenchServerClient):
+    """
+    Used by Pbench Server functional tests to get admin access
+    on OIDC server.
+    """
+    return OIDCAdmin(server_url=server_client.endpoints["openid"]["server"])
+
 
-    response = server_client.post(
-        API.REGISTER,
-        json={
-            "username": "tester",
-            "first_name": "Test",
-            "last_name": "User",
-            "password": "123456",
-            "email": "tester@gmail.com",
-        },
-        raise_error=False,
+@pytest.fixture(scope="session")
+def register_test_user(oidc_admin: OIDCAdmin):
+    """Create a test user for functional tests."""
+    response = oidc_admin.create_new_user(
+        username=USERNAME,
+        email=EMAIL,
+        password=PASSWORD,
+        first_name=FIRST_NAME,
+        last_name=LAST_NAME,
     )
 
     # To allow testing outside our transient CI containers, allow the tester
     # user to already exist.
     assert (
-        response.ok or response.status_code == HTTPStatus.FORBIDDEN
+        response.ok or response.status_code == HTTPStatus.CONFLICT
     ), f"Register failed with {response.json()}"
 
 
 @pytest.fixture
 def login_user(server_client: PbenchServerClient, register_test_user):
     """Log in the test user and return the authentication token"""
-    server_client.login("tester", "123456")
+    server_client.login(USERNAME, PASSWORD)
```

Comment:
Ah: that's what's happening. `server_client` is a *session* scope, and you removed the `yield` and `logout` from the fixture.

Restore the `yield` and just clear `server_client.auth_token` afterwards, and you'll restore the previous "per test case" login, which allows the non-login test cases to run without hacks.